### PR TITLE
Add abstractions and Guid for TapiCall

### LINKED ISSUE #1: minor changes
Hi! nice library!

Just a remark, in the codeplex site, the url to the moved github is pointing to https://github.com/markjulmar/atapi/ and this site is the C/C++ wrapper.

Do you accept pull request? I my current project we add some minor features:
* Abstractions for the main entities, such as TapiLine, TapiCall, TapiAddress. In order to allow mocking those entities in unit tests.
* Added a GUID in the calls to identify it each call as unique.

I can send you a pull request if you want. 

Regards.

### PR DESCRIPTION
**Pull request by vfabregat**
Added a Guid, generated in the TapiCall constructor to allow identify each instance call.
Added interfaces to TapiCall, TapiAddress, CallFeatureSet

This pull request resolve #1

### CODE REVIEW

**Comment by vfabregat** [comment]
Hi, @markjulmar do you have any news on this? do you need help to fix something in the code?

**Comment by mguma** [comment]
Hi, I am also interested in the changes within this pull request.

File: src/src/Interfaces/ITapiAddress.cs [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+        /// <summary>
+        /// The numeric address ID
+        /// </summary>
+        int Id { get; }
+
+        /// <summary>
+        /// The <see>TapiLine</see> associated with the address.
+        /// </summary>
+        TapiLine Line { get; }
```
Comment (markjulmar) [inline src/src/Interfaces/ITapiAddress.cs]: Should these be using the abstractions too? What's the overall purpose of the abstraction given that you can't ever actually _create_ one on your own? For unit tests?

File: src/src/TapiLine.cs [hunk 2]
```diff
@@ -1187,7 +1187,7 @@ public void Close()
             // Close all the calls on the line -- not we don't drop them.
             foreach (TapiAddress addr in _addresses)
             {
-                TapiCall[] calls = addr.Calls;
+                TapiCall[] calls = (TapiCall[])addr.Calls;
```
Comment (markjulmar) [inline src/src/TapiLine.cs]: You shouldn't need this case .. the `foreach` will do the cast for you if you strongly type it. In addition, if the properties/methods are available on the abstraction, then these casts aren't needed.

File: src/src/Interfaces/ITapiAddress.cs (hunk 1 above)
Comment (vfabregat) [reply src/src/Interfaces/ITapiAddress.cs]: Yes, you're right, should be using the abstractions. The purpose is to allow mocking it for unit testing. I'll fix this and send the update.

File: src/src/TapiLine.cs (hunk 2 above)
Comment (vfabregat) [reply src/src/TapiLine.cs]: You're correct, I'll fix it.

**Pull request merged**